### ISSUE TITLE
The match option for python is incorrect

### ISSUE THREAD

**Issue by lsrzj** [firasdib/Regex101#1692]
## Bug Description
Python's match function is anchored at the beginning of the string, so, if you choose the match option and try to match something from the 2nd character to the end, it shouldn't highlight anything

### Reproduction steps
Suppose you're trying to match the end of a timestamp in ISO format '2021-12-31 13:20:30-03' and you put the following regex pattern to match r':[0-9]{2}[+-]' regex101 will highlight the match, while python re.match function will return None.

### Expected Outcome
Regex101 shouldn't highlight and reflect the same behaviour as Python. Instead it should be added the option search that should map to Python's re.search function that scans the entire string without anchoring at the beginning of the string.

### Browser
MS Edge for linux

### OS
Linux Mint 20.04

**Comment by Nexuscompute** [comment]
> ## Bug Description
> Python's match function is anchored at the beginning of the string, so, if you choose the match option and try to match something from the 2nd character to the end, it shouldn't highlight anything
> 
> ### Reproduction steps
> Suppose you're trying to match the end of a timestamp in ISO format '2021-12-31 13:20:30-03' and you put the following regex pattern to match r':[0-9]{2}[+-]' regex101 will highlight the match, while python re.match function will return None.
> 
> ### Expected Outcome
> Regex101 shouldn't highlight and reflect the same behaviour as Python. Instead it should be added the option search that should map to Python's re.search function that scans the entire string without anchoring at the beginning of the string.
> 
> ### Browser
> MS Edge for linux
> 
> ### OS
> Linux Mint 10.0

**Comment by firasdib** [comment]
Not quite sure what you're asking, but I don't see any point in limiting the behavior to the `match` functionality of python. If you want to anchor your regex, use `^` and `$`.

**Comment by lsrzj** [comment]
Python re library is confusing, so it's a little bit hard to Express the problem, I'll try to clarify. In python there are two methods inside re class, with the same objective, match patterns. One of them is re.match and the other one is re.serach, regex101 has two options match or replace. I'm suggesting to add one more option to make regex101 100% compatible with python's re.match and re.search

From the python's documentation:
re.match(string[, pos[, endpos]])[¶](https://docs.python.org/3/library/re.html#re.Pattern.match)
If zero or more characters at the beginning of string match this regular expression, return a corresponding [match object](https://docs.python.org/3/library/re.html#match-objects). Return None if the string does not match the pattern; note that this is different from a zero-length match.

[](https://docs.python.org/3/library/re.html#re.Pattern.search)Pattern.search(string[, pos[, endpos]])
Scan through string looking for the first location where this regular expression produces a match, and return a corresponding [match object](https://docs.python.org/3/library/re.html#match-objects). Return None if no position in the string matches the pattern; note that this is different from finding a zero-length match at some point in the string.

The optional second parameter pos gives an index in the string where the search is to start; it defaults to 0. This is not completely equivalent to slicing the string; the '^' pattern character matches at the real beginning of the string and at positions just after a newline, but not necessarily at the index where the search is to start.

The optional parameter endpos limits how far the string will be searched; it will be as if the string is endpos characters long, so only the characters from pos to endpos - 1 will be searched for a match. If endpos is less than pos, no match will be found; otherwise, if rx is a compiled regular expression object, rx.search(string, 0, 50) is equivalent to rx.search(string[:50], 0).

The optional pos and endpos parameters have the same meaning as for the [search()](https://docs.python.org/3/library/re.html#re.Pattern.search) method.

So there are differences about the behavior from re. match and re.search, and what regex101 is reproducing in the match option is the search behavior, either you could rename the option in regex101 to search or you could rename it to search and create another option called match that maps to re.match behavior. I Hope this comment solved any remaining doubts.

**Comment by firasdib** [comment]
@lsrzj I understand, but match is a subset of search, thus, I see no reason to implement match.

**Comment by lsrzj** [comment]
The suggestion came, because I use regex101 to validate all regex I make, and as I saw the option match in regex101, I found it weird by the fact that python's re.match wasn't behaving the way regex101 does and was returning None instead of a match. Searching on google I found the reason and I noticed that I should use re.search instead. I know I should have done the RTFM thing, but I use regex101 as a trustable source of information and it was a little gotcha for me.
